src/main/java/org/opensearch/ad/transport/AnomalyResultTransportAction.java

```diff
@@ -622,21 +623,26 @@ private void handleQueryFailure(Exception exception, ActionListener<AnomalyResul
 
     /**
      * Convert a query related exception to EndRunException
+     *
+     * These query exception can happen during the starting phase of the OpenSearch
+     * process.  Thus, set the stopNow parameter of these EndRunException to false
+     * and confirm the EndRunException is not a false positive.
+     *
      * @param exception Exception
      * @param adID detector Id
      * @return the converted exception if the exception is query related
      */
     private Exception convertedQueryFailureException(Exception exception, String adID) {
         if (ExceptionUtil.isIndexNotAvailable(exception)) {
-            return new EndRunException(adID, TROUBLE_QUERYING_ERR_MSG + exception.getMessage(), true).countedInStats(false);
+            return new EndRunException(adID, TROUBLE_QUERYING_ERR_MSG + exception.getMessage(), false).countedInStats(false);
         } else if (exception instanceof SearchPhaseExecutionException && invalidQuery((SearchPhaseExecutionException) exception)) {
             // This is to catch invalid aggregation on wrong field type. For example,
```

Comment:
If we change to `false`, can we still catch the invalid aggregation and stop the detector?

Reply:
Yes, we will.

Reply:
As we have changed the `EndRunException` as false here, can you move this comment to that place where we catch invalid aggregation and stop detector? or we can update this comment to make it clear.

Reply:
This is the place to catch invalid aggregation.  We stop detector on job scheduler class.  Everything is the same except we don't stop immediately.